Lib/logging/__init__.py

```diff
@@ -185,14 +185,29 @@ def currentframe():
 
 
 def _checkLevel(level):
-    if isinstance(level, int):
-        rv = level
-    elif str(level) == level:
-        if level not in _nameToLevel:
-            raise ValueError("Unknown level: %r" % level)
-        rv = _nameToLevel[level]
-    else:
-        raise TypeError("Level not an integer or a valid string: %r" % level)
+    """Check parameter against all defined values. Return NOTSET if invalid.
+
+    Since all logging.$level() functions choose to emit based on
+    numeric comparison, a default of ERROR would be more friendly.
+    """
+    rv = NOTSET
+    try:
+        if level in _nameToLevel:
+            rv = _nameToLevel[level]
+        elif level in _levelToName:
+            rv = level
+        else:
+        #FIXME - test harness injects '+1',  so tolerating 
+        # arbitrary integers is expected behavior. Why?
+        #    raise ValueError
+            rv = int(level)
```

Comment:
This makes `'10'` and `12.3` be accepted now.

Reply:
if it can be reduced to an integer than it's perfectly "valid". Otherwise we should strictly check for defined levels. Which apparently is not desired.

Reply:
This should be discussed on the bug tracker. If this is desired, this needs to be documented.